src/expr/src/scalar/func/impls/interval.rs

```diff
@@ -28,13 +28,10 @@ sqlfunc!(
     #[sqlname = "ivtotime"]
     fn cast_interval_to_time(mut i: Interval) -> NaiveTime {
         // Negative durations have their HH::MM::SS.NS values subtracted from 1 day.
-        if i.duration < 0 {
-            i = Interval::new(0, 86400, 0)
+        if i.duration_as_chrono() < chrono::Duration::zero() {
```

Comment:
Since time doesn't contain days, we can just drop the days. I have the following tests in `interval.stl`

```
query T
SELECT INTERVAL '6 days'::time;
----
00:00:00

query T
SELECT INTERVAL '7 months'::time;
----
00:00:00
```

Which matches the behavior of postgres:
```
postgres=# SELECT INTERVAL '6 days'::time;
   time   
----------
 00:00:00
(1 row)

postgres=# SELECT INTERVAL '7 months'::time;
   time   
----------
 00:00:00
(1 row)
```